src/magicbox/SuggestionsManager.ts

```diff
@@ -1,17 +1,27 @@
-import { defaults, each, indexOf } from 'underscore';
+import { defaults, each, indexOf, find } from 'underscore';
 import { IQuerySuggestSelection, OmniboxEvents } from '../events/OmniboxEvents';
 import { Component } from '../ui/Base/Component';
 import { $$, Dom } from '../utils/Dom';
 import { InputManager } from './InputManager';
 import { ResultPreviewsManager } from './ResultPreviewsManager';
 import { QueryProcessor, ProcessingStatus } from './QueryProcessor';
 
+export interface ISuggestionField {
+  name: string;
+  value: string;
+}
+
+export interface ISuggestionCategoryField extends ISuggestionField {
+  categoryFieldDelimitingCharacter: string;
+}
+
 export interface Suggestion {
   text?: string;
   index?: number;
   html?: string;
   dom?: HTMLElement;
   separator?: string;
+  field?: ISuggestionField;
```

Comment:
I think the interface should instead support a `queryExpression` or a `filter` that should be applied, instead of a field.

And the filter would be calculated not by the `QuerySuggestPreview` but by the piece of code that is actually pushing the suggestions (in this case, `FacetValueSuggestions`)

Otherwise it's very limiting, and the code that is "pushing" the suggestions should have more control that it does now about "how" the query should be built.

Reply:
Yes, seems nothing else than `value` is used from the field. Seems extracting the string would be more simple and extensible

Reply:
Alright, I'll replace it with `queryExpression` then.